### PR TITLE
Add force flag for set override and update CNI schema

### LINKED ISSUE #18561: `istioctl manifest` does not work with CNI
```
$ istioctl manifest generate --set profile=demo --set cni.enabled=true --set cni.components.cni.enabled=true --set values.cni.cniBinDir=/var/lib/cni/bin --set values.cni.cniConfDir=/var/run/multus/cni/net.d

bad path=value (values.cni.cniBinDir=/var/lib/cni/bin): unknown field "cni" in v1alpha1.Values
```

This is istioctl from 1.4:
```
$ istioctl version --short=false --remote=false
version.BuildInfo{Version:"69dffe00dec378e2a9a24e42bc36746f26f3c9a6", GitRevision:"69dffe00dec378e2a9a24e42bc36746f26f3c9a6", GolangVersion:"go1.13.3", BuildStatus:"Clean", GitTag:"1.4.0-beta.1-19-g69dffe00d"}
```

### PR DESCRIPTION
**Pull request by richardwxn**
resolve: https://github.com/istio/istio/issues/18561

changes include:
1. add force flag in set override logic, so user can apply --force to ignore validation error for values schema.
2. update outdated values schema for CNI part.

### CODE REVIEW

File: pkg/apis/istio/v1alpha1/values_types.proto [hunk 1]
```diff
@@ -52,6 +52,26 @@ message ArchConfig {
 message CNIConfig {
   // Controls whether CNI is enabled.
   google.protobuf.BoolValue enabled = 1;
+
```
Comment (ostromart) [inline pkg/apis/istio/v1alpha1/values_types.proto]: Add a test to check that bad CNI field causes error, good one passes?

Comment (richardwxn) [reply pkg/apis/istio/v1alpha1/values_types.proto]: added unit test here, would create separate PR for fully validation e2e test

**Comment by richardwxn** [comment]
not sure why the gen check keeps failing even though there is no local update when run make gen..

**Comment by jmazzitelli** [comment]
@richardwxn now you have conflicts :-/

**Comment by ostromart** [comment]
@richardwxn gencheck is probably failing because you have to do make clean gen. Give that a try.
Otherwise lgtm.

**Review by ostromart — APPROVED** [review]

**Comment by jmazzitelli** [comment]
Are these messages about conflicts an indication of something wrong in the 1.4 branch?

**Comment by richardwxn** [comment]
@jmazzitelli actually not, the conflicts come from test data output, see https://github.com/istio/istio/issues/18564.  I am fixing that, hopefully the output manifests order would be stable after that.

**Pull request merged**